### PR TITLE
ASIO: Example tests integration

### PR DESCRIPTION
**Pull request by gabsuren**

### CODE REVIEW

File: .github/workflows/target-test.yml [hunk 1]
```diff
@@ -90,25 +89,45 @@ jobs:
       matrix:
         idf_ver: ["latest"]
         idf_target: ["esp32", "esp32s2"]
-        example: ["asio_chat", "async_request", "socks4", "ssl_client_server", "tcp_echo_server", "udp_echo_server"]
-
+        example: ["asio_chat", "tcp_echo_server", "udp_echo_server", "ssl_client_server"]
```
Comment (david-cermak) [inline .github/workflows/target-test.yml:92]: Question: could we still build the other examples? Even if there's no actual test, it's good I think to have it at least the build step for all of them.

File: .github/workflows/target-test.yml [hunk 2]
```diff
@@ -90,25 +89,45 @@ jobs:
...
     runs-on: ubuntu-20.04
     container: espressif/idf:${{ matrix.idf_ver }}
     steps:
       - name: Checkout esp-protocols
         uses: actions/checkout@v3
         with:
           submodules: recursive 
-          path: esp-protocols
-      - name: Build ${{ matrix.example }} with IDF-${{ matrix.idf_ver }} for ${{ matrix.idf_target }}
+      - name: Build ${{ matrix.example }} with IDF-${{ matrix.idf_ver }} for ${{ matrix.idf_target }} for ${{ matrix.example }}
```
Comment (david-cermak) [inline .github/workflows/target-test.yml:100]: Nitpick: Repeats `Build ${{ matrix.example }} ... for ${{ matrix.example }}`

File: .github/workflows/target-test.yml [hunk 3]
```diff
@@ -90,25 +89,45 @@ jobs:
...
+        working-directory: components/asio/examples/${{ matrix.example }}
         env:
           IDF_TARGET: ${{ matrix.idf_target }}
         shell: bash
         run: |
           . ${IDF_PATH}/export.sh
-          cd $GITHUB_WORKSPACE/esp-protocols/components/asio/examples/${{ matrix.example }}
-          idf.py set-target ${{ matrix.idf_target }}
+          test -f sdkconfig.ci && cat sdkconfig.ci >> sdkconfig.defaults || echo "No sdkconfig.ci"
+          idf.py set-target ${{ matrix.idf_target }}
```
Comment (david-cermak) [inline .github/workflows/target-test.yml:108]:
```suggestion
          idf.py set-target ${{ matrix.idf_target }}
```

**Review by david-cermak — APPROVED** [review]
LGTM, but left some nitpicks

**Comment by gabsuren** [comment]
@david-cermak thanks for the review! Updated changes accordingly

**Review by david-cermak — APPROVED** [review]

**Pull request merged**